test/client/fixtures/sandbox/node/window-test.js

```diff
@@ -76,18 +76,86 @@ test('parameters passed to the native function in its original form', function (
         checkNativeFunctionArgs('setTimeout', 'setTimeout', window);
     }
 
-    if (window.history.pushState && window.history.replaceState) {
-        checkNativeFunctionArgs('pushState', 'historyPushState', window.history);
-        checkNativeFunctionArgs('replaceState', 'historyReplaceState', window.history);
-    }
-
     var documentFragment = document.createDocumentFragment();
 
     checkNativeFunctionArgs('querySelector', 'documentFragmentQuerySelector', documentFragment);
     checkNativeFunctionArgs('querySelectorAll', 'documentFragmentQuerySelectorAll', documentFragment);
     checkNativeFunctionArgs('dispatchEvent', 'windowDispatchEvent', window);
 });
 
+if (window.history.replaceState) {
+    asyncTest('replace history state with a various types of url', function () {
+        function SomeClass () {
+            this.url = 'url';
+        }
+
+        SomeClass.prototype.toString = function () {
+            return this.url;
+        };
+
+        var urlValues = [
+            'string',
+            12345,
+            true,
+            false,
+            NaN,
+            Infinity,
+            null,
+            void 0,
+            new SomeClass()
+        ];
+
+        var iframe  = document.createElement('iframe');
+
+        iframe.setAttribute('src', window.QUnitGlobals.getResourceUrl('../../../data/history/iframe.html'));
+
+        window.QUnitGlobals.waitForIframe(iframe)
+            .then(function () {
+                var iframeWindow        = iframe.contentWindow;
+                var iframeHammerhead    = iframeWindow['%hammerhead%'];
+                var iframeHistory       = iframeWindow.history;
+                var iframeNativeMethods = iframeHammerhead.nativeMethods;
+                var iframeLocation      = iframeWindow.location;
+                var baseUrl             = 'http://' + location.host + '/some/path';
+
+                iframeHammerhead.get('./utils/url-resolver').updateBase(baseUrl, iframe.contentDocument);
+                iframeHammerhead.get('./utils/destination-location')
+                    .forceLocation('http://' + iframeLocation.host + '/sessionId/' + baseUrl);
+
+                var testUrl = function (url, fnName, nativeFn) {
```

Comment:
Les't change the second function parameter to `fn` and pass overriden function respectively.